internal/policy/container/has_unique_tag.go

```diff
@@ -57,13 +57,44 @@ func (p *hasUniqueTagCheck) Validate(ctx context.Context, imgRef image.ImageRefe
 }
 
 func (p *hasUniqueTagCheck) getDataToValidate(ctx context.Context, image string) ([]string, error) {
-	options := []crane.Option{
-		crane.WithContext(ctx),
-		crane.WithAuthFromKeychain(authn.PreflightKeychain(ctx, authn.WithDockerConfig(p.dockercfg))),
-		option.RetryOnceAfter(5 * time.Second),
+	repo, err := name.NewRepository(image)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse image name: %v", err)
 	}
 
-	return crane.ListTags(image, options...)
+	options := []remote.Option{
+		remote.WithContext(ctx),
+		remote.WithAuthFromKeychain(authn.PreflightKeychain(ctx, authn.WithDockerConfig(p.dockercfg))),
+		remote.WithRetryBackoff(remote.Backoff{
+			Duration: 5 * time.Second,
+			Factor:   1.0,
+			Jitter:   0.1,
+			Steps:    2,
+		}),
```

Comment:
Do we see any benefit in restricting the number of tags we want via the [remote.WithPageSize](https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/options.go#L288-L297) option? If I'm reading correctly, I see the default set to [1000](https://github.com/google/go-containerregistry/blob/main/pkg/v1/remote/options.go#L108), so we're still doing a large-ish pull for something we can evaluate with a smaller pull.

Reply:
cc @acornett21

Reply:
We could probably add this now that we aren't getting multiple pages. I'm not sure what value we would want, technically we only care if there is `one more tag then latest` , but going with just `WithPageSize(2)` seems risky.

https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/4b330505988d594b55cbb9e022b5669d7f19e179/internal/policy/container/has_unique_tag.go#L74

Reply:
For me, anything smaller than 10 seems fit for purpose while also reducing the query.  Sound reasonable enough?

Also, worth adding a log line here at DEBUG/TRACE verbosity, listing "which tags were returned"?

Reply:
@caxu-rh Any concern with reducing the page size? You can decide on the value here. Just add the log line so we know what tags were pulled in the event of an issue and I think this should be set.